src/components/TransactionForms/CustomForms/SmartContract/index.tsx

```diff
@@ -69,17 +69,39 @@ const parseFunctionArguments = (
   delete data.function;
 
   const parsedArgsString = getParsedArgumentsString(args, abi);
-
   if (scType === 1) {
     // Deploy
     if (metadata?.length === 0) {
       return;
     }
+
+    let contractBinaryAndParams = metadata.split('@').slice(0, 3).join('@');
+
+    if (parsedArgsString?.length > 0) {
+      contractBinaryAndParams += `@${parsedArgsString}`;
+    }
+
+    setMetadata(contractBinaryAndParams);
+
+    return;
+  } else if (scType === 2) {
+    // Upgrade
```

Comment:
instead of comment, could split this function part into a new function like upgradeMetadata